core/meta/src/TClass.cxx

```diff
@@ -5464,20 +5464,15 @@ TClass *TClass::LoadClassCustom(const char *requestedname, Bool_t silent)
 
 void TClass::LoadClassInfo() const
 {
+   // Return early if class info cannot be loaded or is already loaded.
+   if (!fCanLoadClassInfo) return;
+
    R__LOCKGUARD(gInterpreterMutex);
 
    // If another thread executed LoadClassInfo at about the same time
    // as this thread return early since the work was done.
    if (!fCanLoadClassInfo) return;
 
-   // If class info already loaded then do nothing.  This can happen if the
-   // class was registered by a dictionary, but the info came from reading
-   // the pch.
-   // Note: This check avoids using AutoParse for classes in the pch!
-   if (fClassInfo) {
```

Comment:
Thanks for explaining @amadio And @pcanal *bad* naming :-) "can load" is the negation of "have loaded"?